system/lib/libc/emscripten_pthread.c

```diff
@@ -24,12 +25,28 @@ EM_JS(void, initPthreadsJS, (void), {
   PThread.initRuntime();
 })
 
+static void init_file_lock(FILE *f)
```

Comment:
Can you add a comment along that lines of "This code is copied from musl/src/thread/pthread_create.c.  Keep in sync"

Same below where init_file_lock is called.